src/main/java/com/atlassian/jira/cloud/jenkins/listeners/ConfigurationSaveableListener.java

```diff
@@ -1,41 +1,140 @@
 
-    private static final Logger LOGGER =
-            Logger.getLogger(ConfigurationSaveableListener.class.getName());
+    private PluginConfigApi pluginConfigApi;
+    private SecretRetriever secretRetriever;
+    private XmlUtils xmlUtils;
 
-    private transient PluginConfigApi pluginConfigApi;
+    @Inject
+    public void setSecretRetriever(final SecretRetriever secretRetriever) {
+        if (secretRetriever == null) {
+            throw new IllegalArgumentException("SecretRetriever cannot be null");
+        }
+        this.secretRetriever = secretRetriever;
+    }
 
     @Inject
     public void setConnectionDataApi(final PluginConfigApi pluginConfigApi) {
+        if (pluginConfigApi == null) {
+            throw new IllegalArgumentException("PluginConfigApi cannot be null");
+        }
         this.pluginConfigApi = pluginConfigApi;
     }
 
+    @Inject
+    public void setXmlUtils(final XmlUtils xmlUtils) {
+        if (xmlUtils == null) {
+            throw new IllegalArgumentException("XmlUtils cannot be null");
+        }
+        this.xmlUtils = xmlUtils;
+    }
+
     @Override
     public void onChange(final Saveable saveable, final XmlFile file) {
-        if (!(saveable instanceof com.atlassian.jira.cloud.jenkins.config.JiraCloudPluginConfig)) {
-            return;
+        if (saveable instanceof com.atlassian.jira.cloud.jenkins.config.JiraCloudPluginConfig) {
+            sendPluginConfigData(file);
+        }
+    }
+
+    void sendPluginConfigData(final XmlFile file) {
+        try {
+            Element rootElement = this.xmlUtils.parseXmlFile(file);
+            List<PluginSiteData> siteDataList = createSiteDataList(rootElement);
+            String autoBuildsRegex = this.xmlUtils.extractXmlValue(rootElement, "autoBuildsRegex");
+            boolean autoBuildsEnabled =
+                    this.xmlUtils.extractBooleanValue(rootElement, "autoBuildsEnabled");
+            boolean autoDeploymentsEnabled =
+                    this.xmlUtils.extractBooleanValue(rootElement, "autoDeploymentsEnabled");
+            String autoDeploymentsRegex =
+                    this.xmlUtils.extractXmlValue(rootElement, "autoDeploymentsRegex");
+
+            for (PluginSiteData data : siteDataList) {
+                Optional<String> maybeSecret =
+                        secretRetriever.getSecretFor(data.getCredentialsId());
+                maybeSecret.ifPresent(
+                        secret ->
+                                sendDataToPluginApi(
+                                        data,
+                                        autoBuildsRegex,
+                                        autoBuildsEnabled,
+                                        autoDeploymentsEnabled,
+                                        autoDeploymentsRegex,
+                                        secret));
+            }
+        } catch (Exception e) {
```

Comment:
What are your thoughts on doing something like:

```
} catch (ParserConfigurationException | SAXException | IOException e) {
    logger.log(Level.SEVERE, "Error processing plugin configuration data", e);
}
```

instead of just a generic exception? Will make it easier to diagnose any issues should they occur.

Reply:
Done